File: aula5/decorator.js
function f(){
  console.log('Execuntado f');

  return 'f';
}


function decorador(funcao){
  function calcularTempo(){
    console.log('Iniciando calcular Tempo');
    var tempoInicial=new Date();
    var retornoDeF=funcao.apply(funcao,arguments);
    var tempoFinal=new Date();
    var tempoFinalMs = tempoFinal.getTime();
    var tempoInicialMs = tempoInicial.getTime();
    console.log(tempoFinalMs- tempoInicialMs);
    console.log('Finalizando calcular Tempo');
    return retornoDeF;
  }

  return calcularTempo;

}

f=decorador(f);

function somar(a,b){
  console.log('Execuntado Soma');
  return a+b;
}

console.log(somar([1, 2]));

somar=decorador(somar);

console.log(f());
console.log(somar(2,3));

File: decorator.py
import time


def decorador(funcao):
    def calcularTempo(*arg):
        tempo_inicial = time.time()
        r = funcao(*arg)
        tempo_final = time.time();
        print(tempo_final - tempo_inicial)
        return r

    return calcularTempo

@decorador
def somar(a, b):
    return a + b


# somar = decorador(somar)

print(somar(1, 2))
